### PR TITLE
first version

### PR DESCRIPTION
**Pull request by Yu1986**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by Yu1986** [comment]
@duynguyen @icaraps

File: README.md [hunk 1]
```diff
@@ -1,29 +1,28 @@
...
-
-Demo: https://adobeio-codelabs-template-adobedocs.project-helix.page/ 
-
-Make sure to push your changes before sharing the URL.
-
-## Contribute to Adobe I/O CodeLabs
-
-Once your codelab is ready, submit a PR to add it to the index at https://github.com/AdobeDocs/adobeio-codelabs/blob/gh-pages/actions/index.json  
+In order to demo how to using journaling API to consume events, we provide an end to end solution in this codelab, 
+- Event provider - we need to create an event provider to automatically generate events sending to Jouranling API or if you already have event provider you could skip this step
```
Comment (icaraps) [inline README.md]: typo Jouranling

File: README.md [hunk 2]
```diff
@@ -1,29 +1,28 @@
...
+- Event consumer - which is the main demo part of this codelab, we create another Project Firefly headless app to create cron jobs with alarms, we set up recurring jobs to pull from journalling API every x mins and write into project firefly storage.
```
Comment (icaraps) [inline README.md]: typo journalling

File: lessons/lesson1.md [hunk 3]
```diff
@@ -1,5 +1,69 @@
...
+In this lesson, we will do the follow steps:
+- Create an event provider using Project Firefly template
+- Register the App as event provider 
+- Fire Events
+- Scheduling cron jobs with alarms
 
-Next lesson: [Creating your app](lesson2.md)
\ No newline at end of file
+### Create an event provider using Project Firefly template
+In this codelab, to provide an end to end solution, we need to have an event provider generating tons of events sending to journaling API, and the events provider need to be configurable to send x events/min to help user to test. so we choose to use OpenWhisk Alarms Package in Fierfly application to create
```
Comment (icaraps) [inline lessons/lesson1.md]: typo Fierfly

File: lessons/lesson3.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+## Lesson 3: End to end test
+In the previous lessons, we have setup two firefly apps:
+- Event provider that automatically generating events 
+- Event consumer that automatically pulling from journaling api and write to firefly storage
+
+Let’s first set in `manifest.yml` and try the `/whisk.system/alarms/interval` feed of the OpenWhisk Alarm Package to fire trigger events on an interval base schedule. To see the effect instantly, we will make it run every minute. You will need a trigger set up with the `/whisk.system/alarms/interval` feed, and a rule to wire this trigger to the `publish-event` or `consume-event` action created earlier.
```
Comment (icaraps) [inline lessons/lesson3.md]: Mhh I don't understand this part ? What are you supposed to do in `manifest` ?

**Review by icaraps — COMMENTED** [review]
LGTM  just left a few comments

File: lessons/lesson3.md (hunk 4 above)
Comment (Yu1986) [reply lessons/lesson3.md]: if people would like to use the alarm package to automatically generate event and pulling event

**Comment by Yu1986** [comment]
revise and merge it

**Pull request merged**